engine/src/main/java/org/destinationsol/world/generators/FeatureGenerator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.world.generators;
+
+/**
+ * This class represents the abstract version of any feature which will populate the game's SolSystems, such as Planets,
```

Comment:
Will the subclasses _create_ the planets/mazes, or will they _be_ the planets/mazes? It could be that I'm not understanding how you're using the word "feature", but it sounds like the subclasses will be instances of features, meaning that the subclasses would be Planet instances.

Reply:
@IsaacLic Yeah I think the word 'be' is more correct. I had meant create as in place themselves into the world. Meaning each SolarSystemGenerator will pick whichever FeatureGenerators it wants to use, and then those FeatureGenerators will be capable of placing an instance of themselves into the game